text_to_image/backend_pytorch.py

```diff
@@ -0,0 +1,415 @@
+        prompt_embeds = prompt_embeds.view(
+            bs_embed * num_images_per_prompt, seq_len, -1
+        )
+
+        if do_classifier_free_guidance:
+            # duplicate unconditional embeddings for each generation per prompt, using mps friendly method
+            seq_len = negative_prompt_embeds.shape[1]
+
+            if pipe.text_encoder_2 is not None:
+                negative_prompt_embeds = negative_prompt_embeds.to(
+                    dtype=pipe.text_encoder_2.dtype, device=device
+                )
+            else:
+                negative_prompt_embeds = negative_prompt_embeds.to(
+                    dtype=pipe.unet.dtype, device=device
+                )
+
+            negative_prompt_embeds = negative_prompt_embeds.repeat(
+                1, num_images_per_prompt, 1
+            )
+            negative_prompt_embeds = negative_prompt_embeds.view(
+                batch_size * num_images_per_prompt, seq_len, -1
+            )
+
+        pooled_prompt_embeds = pooled_prompt_embeds.repeat(
+            1, num_images_per_prompt
+        ).view(bs_embed * num_images_per_prompt, -1)
+        if do_classifier_free_guidance:
+            negative_pooled_prompt_embeds = negative_pooled_prompt_embeds.repeat(
+                1, num_images_per_prompt
+            ).view(bs_embed * num_images_per_prompt, -1)
+        return (
+            prompt_embeds,
+            negative_prompt_embeds,
+            pooled_prompt_embeds,
+            negative_pooled_prompt_embeds,
+        )
+
+    def predict(self, inputs):
+        images = []
+        with torch.no_grad():
+            for prompt in inputs:
+                assert isinstance(prompt, dict)
+                text_input = prompt["input_tokens"]
+                text_input_2 = prompt["input_tokens_2"]
+                refiner_text_input = prompt["refiner_input_tokens"]
+                refiner_text_input_2 = prompt["refiner_input_tokens_2"]
+                latents_input = prompt["latents"].to(self.dtype)
+                (
+                    prompt_embeds,
+                    negative_prompt_embeds,
+                    pooled_prompt_embeds,
+                    negative_pooled_prompt_embeds,
+                ) = self.encode_tokens(
+                    self.pipe,
+                    text_input,
+                    text_input_2,
+                    negative_prompt=self.negative_prompt_tokens,
+                    negative_prompt_2=self.negative_prompt_tokens_2,
+                )
+                image = self.pipe(
+                    prompt_embeds=prompt_embeds,
+                    negative_prompt_embeds=negative_prompt_embeds,
+                    pooled_prompt_embeds=pooled_prompt_embeds,
+                    negative_pooled_prompt_embeds=negative_pooled_prompt_embeds,
+                    guidance_scale=self.guidance,
+                    num_inference_steps=self.steps,
+                    output_type="latent",
```

Comment:
Since we do not use refiner network, this will need to change to go through VAE module right?

Reply:
Done